tests/rptest/scale_tests/high_throughput_test.py

```diff
@@ -0,0 +1,928 @@
+            finally:
+                producer.stop()
+                producer.wait(timeout_sec=600)
+        finally:
+            self.free_preallocated_nodes()
+
+        # Once some segments are generated, configure the topic to use more
+        # realistic sizes.
+        retention_bytes = int(self.scaled_data_bps * 60 * 60 * 6 /
+                              self.scaled_num_partitions)
+        self.rpk.alter_topic_config(self.topic_name, 'segment.bytes',
+                                    self.regular_segment_size)
+        self.rpk.alter_topic_config(self.topic_name,
+                                    'retention.local.target.bytes',
+                                    retention_bytes)
+
+    def get_node(self, idx: int):
+        node = self.redpanda.nodes[idx]
+        node_id = self.redpanda.node_id(node)
+        node_str = f"{node.account.hostname} (node_id: {node_id})"
+        return node, node_id, node_str
+
+    @cluster(num_nodes=5, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_combo_preloaded(self):
+        """
+        Preloads cluster with large number of messages/segments that are also
+        replicated to the cloud, and then run various different test stages
+        on the cluster:
+        - rolling restart
+        - stop and start single node, various scenarios
+        - isolate and restore a node
+        """
+        self.setup_cluster(segment_bytes=self.small_segment_size,
+                           retention_local_bytes=2 * self.small_segment_size)
+        # Generate a realistic number of segments per partition.
+        self.load_many_segments()
+        try:
+            producer = KgoVerifierProducer(
+                self.test_context,
+                self.redpanda,
+                self.topic_name,
+                msg_size=128 * 1024,
+                msg_count=5 * 1024 * 1024 * 1024 * 1024,
+                rate_limit_bps=self.scaled_data_bps,
+                custom_node=[self.preallocated_nodes[0]])
+            try:
+                producer.start()
+                wait_until(lambda: producer.produce_status.acked > 10000,
+                           timeout_sec=60,
+                           backoff_sec=1.0)
+
+                # Run a rolling restart.
+                self.stage_rolling_restart()
+
+                # Hard stop, then restart.
+                self.stage_stop_wait_start(forced_stop=True, downtime=0)
+
+                # Stop a node, wait for enough time for movement to occur, then
+                # restart.
+                self.stage_stop_wait_start(forced_stop=False,
+                                           downtime=self.unavailable_timeout)
+
+                # Stop a node and wait for really long time to collect a lot
+                # of under-replicated msgs, then restart.
+                # This is not to be run nightly so disabled for now
+                #self.stage_stop_wait_start(forced_stop=False, downtime=60*30)
+
```

Comment:
Not in this PR, but we should flesh out a way to annotate slow tests that get run less frequently. I guess the closest thing we have is checking whether we're running with dedicated nodes, but as we build out different testing schedules this would be a great place to use that kind of info.

Reply:
Agree